### ISSUE TITLE
PINCE keeps being detected and shutdown, also slows game

### ISSUE THREAD

**Issue by awsdert** [korcankaraokcu/PINCE#97]
Playing FFXV but it keeps detecting and shutting down PINCE, it also runs a lot slower with PINCE attached, GameConqueror doesn't have this problem so I believe it's because you're using *.so injection, even capabilities get disabled for it please add an option to use just the standard process_memory_read()/process_memory_write() etc functions - I think that's what the were called, I've use those functions in a poorly programmed app here:

https://github.com/awsdert/gasp

Go ahead and borrow whatever code you need, just get PINCE working with FFXV please

**Comment by l3arnmore** [comment]
if you cant even c+p avalibable code to bypass c+p'd anti-debug routines, just stop. you are embarassing yourself.

**Comment by korcankaraokcu** [comment]
> I believe it's because you're using *.so injection

No, PINCE doesn't use *.so injection in its initial attach routine. It only uses GDB in the background. That's probably the thing FFXV is complaining about. Like @l3arnmore said, it's most likely the anti-debug counter measures. PINCE actually has memory read/write functions that are independent from GDB but for other functionality, it uses GDB.

PINCE attach routine could be rewritten to only use those functions and disable the rest of the functionality but there would be no point to use PINCE with such games then. Simply using GC would be much better for this case unless PINCE gets new features that are independent from GDB.

**Comment by awsdert** [comment]
Okay, then just add the option, as far as the UI goes it shouldn't take anything more than a checkbox that says something like "limited but safe" then has a tool tip that says something like "if pince randomly shuts down during a search on specific games try this with those games"

Better limited functionality than a program that just shuts down without warning because a game sent a kill message

**Comment by awsdert** [comment]
As a side note scanning memory, registers, etc, never needs to attached the whole time, just dump everything to a directory, then detach, then do the search using the dumped stuff, makes it easier to abstract out stuff that hooks & dumps that stuff. This would also remove the need for the process to be stopped when looking at the memory/registers, just briefly stop, dump the area looking at then detach. The time saved by staying attached is not worth it.

**Comment by korcankaraokcu** [comment]
> never needs to attached the whole time

This is not true. There are a few debugging utilities that make use of the GDB session, such as data breakpoints. So, GDB **must** stay active for all functionality to work. You are forgetting that this project is essentially a GDB frontend.
